gm4_heart_canisters/data/gm4_heart_canisters/function/check_heart_canisters.mcfunction

```diff
@@ -1,13 +1,21 @@
-# gets called for every player with at least 1 heart canister tier 1 item
+# gets called for every player with at least 1 heart canister item
 
-scoreboard players set @s[nbt={Inventory:[{count:1,components:{"minecraft:custom_data":{gm4_heart_canister_tier:1b}}}]}] gm4_heart_can 1
-scoreboard players set @s[nbt={Inventory:[{count:2,components:{"minecraft:custom_data":{gm4_heart_canister_tier:1b}}}]}] gm4_heart_can 2
-scoreboard players set @s[nbt={Inventory:[{count:3,components:{"minecraft:custom_data":{gm4_heart_canister_tier:1b}}}]}] gm4_heart_can 3
-scoreboard players set @s[nbt={Inventory:[{count:4,components:{"minecraft:custom_data":{gm4_heart_canister_tier:1b}}}]}] gm4_heart_can 4
-scoreboard players set @s[nbt={Inventory:[{count:5,components:{"minecraft:custom_data":{gm4_heart_canister_tier:1b}}}]}] gm4_heart_can 5
+# tier 1
+scoreboard players set $count_total gm4_heart_can 0
+data modify storage gm4_heart_cannisters:temp heart_cannister_data set value []
+data modify storage gm4_heart_cannisters:temp heart_cannister_data append from entity @s Inventory[{components:{"minecraft:custom_data":{gm4_heart_canister_tier:1b}}}]
+execute store result score $count_add gm4_heart_can run data get entity @s equipment.offhand{components:{"minecraft:custom_data":{gm4_heart_canister_tier:1b}}}.count
+scoreboard players operation $count_total gm4_heart_can += $count_add gm4_heart_can
+function gm4_heart_canisters:count_heart_canisters
+scoreboard players operation $count_total gm4_heart_can < $heart_cannister_cap gm4_heart_can
+scoreboard players operation @s gm4_heart_can += $count_total gm4_heart_can
```

Comment:
This is an abysmal amount of player NBT checks, I'd much rather see 10 `if items` checks in a single mcfunction. And this score is no longer capped at `5`. I would either pick one of these changes:
- heart canisters stack to max 5
- heart canisers can now be spread around in the inventory

If you implement the latter, there is no longer any advantage of making them only stack to 5 imo.

Reply:
Unless I'm misunderstanding how player checks work this shouldn't be more than before, since the data is stored in storage first. I could change it to 20 `if items`. Making them stack to 5 feels like a good way to show players that that is the max amount that work

Reply:
Altough I guess `execute store result score $count gm4_heart_can if items entity @s container.* <heart can>` would work to count them as well, if that's better than looping over a storage

Reply:
Okay yeah, much better. Although the test is still failing, is the cap working?

Reply:
Test was failing since it assumed 6 canisters resulted in no health gain, while it now grants the effect of 5 canisters, I've updated the test